internal/server/routes.go

```diff
@@ -242,11 +252,30 @@ func wrapRoute[Req, Res any](a *API, route route[Req, Res]) func(*gin.Context) e
 			a.t.RouteEvent(c, route.path, Properties{"method": strings.ToLower(route.method)})
 		}
 
-		c.JSON(responseStatusCode(route.method, resp), resp)
+		if r, ok := responseIsRedirect(resp); ok {
+			c.Redirect(http.StatusPermanentRedirect, r.RedirectTo)
+		} else {
+			c.JSON(responseStatusCode(route.method, resp), resp)
+		}
 		return nil
 	}
 }
 
+func responseIsRedirect(resp any) (api.RedirectResponse, bool) {
```

Comment:
I wonder if we can avoid the reflection here. If `api.RedirectResponse` implemented something like:

```go
type redirectResponse interface {
    statusCoder
    RedirectTo() string
}
```

I think that would allow us to use the same pattern as `responseStatusCode`

Reply:
a good suggestion, but I'm not sure that works. the response isn't an interface, it's a concrete type you just don't know what it is because of generics.

Reply:
You'd have to do something like 
```
		var r interface{} = resp
		if rsp, ok := r.(isRedirect); ok {
			c.Redirect(http.StatusPermanentRedirect, rsp.RedirectURL())
		}
```

Reply:
nm this is fine, I've got it looking good.